### PR TITLE
Implement serialization for WitnessSet

### LINKED ISSUE #90: Serialization for WitnessSet
See commented-out code in serialization module. We need it before we can send the signed transaction.

### PR DESCRIPTION
**Pull request by klntsky**
Closes #90

### CODE REVIEW

File: src/Serialization/BigNum.purs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Serialization.BigNum
+  ( convertBigNum
+  ) where
+
+import Prelude
+import Serialization.Types (BigNum)
+import FfiHelpers
+import Data.BigInt as BigInt
+import Data.BigInt (BigInt)
+import Data.Maybe
```
Comment (vvtran) [inline src/Serialization/BigNum.purs:10]: Let's import `Maybe` here

File: src/Serialization/BigNum.purs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module Serialization.BigNum
+  ( convertBigNum
+  ) where
+
+import Prelude
+import Serialization.Types (BigNum)
+import FfiHelpers
```
Comment (vvtran) [inline src/Serialization/BigNum.purs:7]: `import FfiHelpers (MaybeFfiHelper, maybeFfiHelper)`

File: src/Deserialization/FromBytes.purs [hunk 3]
```diff
@@ -31,6 +31,15 @@ instance FromBytes Ed25519KeyHash where
...
 
+instance FromBytes PlutusData where
+  fromBytes = _fromBytesPlutusData maybeFfiHelper
+
+instance FromBytes TransactionUnspentOutput where
+  fromBytes = _fromBytesTransactionUnspentOutput maybeFfiHelper
+
+instance FromBytes TransactionWitnessSet where
+  fromBytes = _fromBytesTransactionWitnessSet maybeFfiHelper
+
```
Comment (vvtran) [inline src/Deserialization/FromBytes.purs:42]: Not part of this PR admittedly, but:
```
import Serialization.Types (Address, DataHash, Ed25519KeyHash, PlutusData, ScriptHash, TransactionHash, TransactionUnspentOutput, TransactionWitnessSet)
import Types.ByteArray (ByteArray)
import Data.Maybe (Maybe(Just, Nothing))
import FfiHelpers (MaybeFfiHelper, maybeFfiHelper)
import Effect (Effect)
import Effect.Exception (throw)
```

File: src/Serialization/BigNum.purs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import Data.BigInt as BigInt
+import Data.BigInt (BigInt)
+import Data.Maybe
+
+convertBigNum :: BigInt -> Maybe BigNum
```
Comment (vvtran) [inline src/Serialization/BigNum.purs:12]: Most (if not all?) of the naming conventions for `convertSomething`  have so far been the "same" thing from foreign data types to/from Purescript types. These two are however different types, is `bigNumFromBigInt` or `bigIntToBigNum` perhaps a bit more descriptive? They are a little long winded however.

File: src/Serialization/WitnessSet.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/* global require exports */
+
+var lib;
+if (typeof BROWSER_RUNTIME != 'undefined' && BROWSER_RUNTIME) {
+    lib = require('@ngua/cardano-serialization-lib-browser');
+} else {
+    lib = require('@ngua/cardano-serialization-lib-nodejs');
+}
```
Comment (vvtran) [inline src/Serialization/WitnessSet.js:8]: Don't have much JS experience but we have this code repeated in a lot of js modules. Should we centralise this?

File: test/Fixtures.purs [hunk 6]
```diff
@@ -0,0 +1,298 @@
+module Test.Fixtures where
```
Comment (vvtran) [inline test/Fixtures.purs:1]: Explicit exports

File: test/Utils.purs [hunk 7]
```diff
@@ -2,11 +2,11 @@ module Test.Utils (unsafeCall, assertTrue, assertTrue_, errMaybe) where
 
 import Control.Alternative (pure)
 import Data.Function (($))
-import Data.Maybe (Maybe, maybe)
+import Data.Maybe (Maybe(..))
```
Comment (vvtran) [inline test/Utils.purs:5]: `Just, Nothing`

File: src/Serialization.purs [hunk 8]
```diff
@@ -3,26 +3,27 @@ module Serialization
...
   , toBytes
   , newBaseAddressFromAddress
-  , newTransactionUnspentOutputFromBytes
-  , newTransactionWitnessSetFromBytes
   ) where
 
 import Data.BigInt as BigInt
-import Data.Maybe (Maybe(Just, Nothing))
+import Data.FoldableWithIndex (forWithIndex_)
+import Data.Maybe (Maybe(..), maybe)
```
Comment (vvtran) [inline src/Serialization.purs:17]: `Just, Nothing`

File: test/Deserialization.purs [hunk 9]
```diff
@@ -5,55 +5,36 @@ import Prelude
...
-  ) as T
-import Types.Value (mkCoin, mkValue, mkNonAdaAssets) as V
-import Types.TransactionUnspentOutput
-  ( TransactionUnspentOutput(TransactionUnspentOutput)
-  ) as T
+import Serialization.WitnessSet as SW
+import Test.Fixtures (addressString, txInputFixture1, txOutputFixture1, utxoFixture1, utxoFixture1', witnessSetFixture1, witnessSetFixture2, witnessSetFixture2Value, witnessSetFixture3, witnessSetFixture3Value, witnessSetFixture4)
+import Test.Utils (errMaybe)
+import Types.Transaction (Bech32(Bech32), TransactionInput, TransactionOutput) as T
+import Types.TransactionUnspentOutput (TransactionUnspentOutput(..)) as T
```
Comment (vvtran) [inline test/Deserialization.purs:27]: Same here.

File: test/Fixtures.purs [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+
+import Data.BigInt as BigInt
+import Data.Maybe (Maybe(..), fromJust)
+import Data.Tuple.Nested ((/\))
+import Data.UInt as UInt
+import Partial.Unsafe (unsafePartial)
+import Types.ByteArray (ByteArray, byteArrayFromIntArrayUnsafe, hexToByteArrayUnsafe)
+import Types.Transaction (Address(..), BaseAddress(..), Bech32(..), Ed25519KeyHash(..), Ed25519Signature(..), NetworkId(..), PaymentCredential(..), PlutusData(..), PublicKey(..), StakeCredential(..), Transaction(..), TransactionHash(..), TransactionInput(..), TransactionOutput(..), TransactionWitnessSet(..), TxBody(..), Vkey(..), Vkeywitness(..)) as T
+import Types.TransactionUnspentOutput (TransactionUnspentOutput(..)) as T
+import Types.Value (Coin(..), CurrencySymbol, TokenName, Value(..)) as T
```
Comment (vvtran) [inline test/Fixtures.purs:13]: bunch of data constructor imports here missing.

File: test/Serialization.purs [hunk 11]
```diff
@@ -2,29 +2,23 @@ module Test.Serialization (suite) where
 
 import Prelude
 
-import Data.BigInt as BigInt
-import Data.Maybe (Maybe(Just, Nothing))
-import Data.Tuple.Nested ((/\))
-import Data.UInt as UInt
+import Data.Maybe (Maybe(..))
```
Comment (vvtran) [inline test/Serialization.purs:5]: Here too.

**Review by vvtran — COMMENTED** [review]
Lgtm, mostly import comments, and a few more specific comments.

**Review by ngua — APPROVED** [review]
LGTM. I'll approve this now so it won't get held up even if you make some changes
